dbms/src/Columns/tests/gtest_column_scatterTo.cpp

```diff
@@ -0,0 +1,233 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Columns/ColumnArray.h>
+#include <Columns/ColumnFixedString.h>
+#include <Columns/ColumnNothing.h>
+#include <Columns/ColumnSet.h>
+#include <Columns/ColumnTuple.h>
+#include <Common/COWPtr.h>
+#include <TestUtils/FunctionTestUtils.h>
+#include <TestUtils/TiFlashTestBasic.h>
+
+namespace DB
+{
+namespace tests
+{
+class TestColumnScatterTo : public ::testing::Test
+{
+public:
+    void compareColumn(const ColumnWithTypeAndName & expected_col_with_type_name,
+                       const ColumnWithTypeAndName & actual_col_with_type_name)
+    {
+        const ColumnPtr expected = expected_col_with_type_name.column;
+        const ColumnPtr actual = actual_col_with_type_name.column;
+        if unlikely (typeid_cast<const ColumnSet *>(expected.get()) || typeid_cast<const ColumnSet *>(actual.get()))
+        {
+            /// ColumnSet compares size only now, since the test ensures data is equal
+            const ColumnSet * expected_set = typeid_cast<const ColumnSet *>(expected.get());
+            const ColumnSet * actual_set = typeid_cast<const ColumnSet *>(actual.get());
+            ASSERT_TRUE(expected_set && actual_set);
+            ASSERT_TRUE(expected_set->size() == actual_set->size());
+            return;
+        }
+        ASSERT_COLUMN_EQ(expected, actual);
+    }
+
+    void doTestWork(ColumnWithTypeAndName & col_with_type_and_name)
+    {
+        auto column_ptr = col_with_type_and_name.column;
+        ASSERT_TRUE(rows == column_ptr->size());
+        MutableColumns scatterTo_res(scattered_num_column);
+        for (size_t i = 0; i < scattered_num_column; i++)
```

Comment:
```suggestion
        for (size_t i = 0; i < scattered_num_column; ++i)
```
from https://clickhouse.com/docs/en/development/style/